x-pack/plugins/ml/public/application/routing/routes/new_job/index_or_search.tsx

```diff
@@ -47,10 +51,10 @@ export const indexOrSearchRoute: MlRoute = {
       mode={MODE.NEW_JOB}
     />
   ),
-  breadcrumbs,
-};
+  breadcrumbs: getBreadcrumbs(application),
+});
 
-export const dataVizIndexOrSearchRoute: MlRoute = {
+export const dataVizIndexOrSearchRouteFactory = (application: ApplicationStart): MlRoute => ({
```

Comment:
If I select Data Visualizer off the top nav, then choose 'Index based', clicking on indexes or saved searches on this page does not take me to the Data Visualizer page:

![image](https://user-images.githubusercontent.com/7405507/87793651-746bbf00-c83d-11ea-89ad-0dfff33659dd.png)

Reply:
Fixed in 0cae8a1.